gst/nnstreamer/elements/gsttensor_split.c

```diff
@@ -594,25 +597,36 @@ gst_tensor_split_set_property (GObject * object, guint prop_id,
       guint i;
       const gchar *param = g_value_get_string (value);
       gchar **strv = g_strsplit_set (param, ",.;/", -1);
+      GArray *tensorseg = split->tensorseg;
+
       split->num_tensors = g_strv_length (strv);
-      split->tensorseg =
-          g_array_sized_new (FALSE, FALSE, sizeof (tensor_dim *),
-          split->num_tensors);
+      if (NULL == tensorseg) {
+        split->tensorseg =
+            g_array_sized_new (FALSE, FALSE, sizeof (tensor_dim *),
+            split->num_tensors);
+
```

Comment:
With @songgot 's opinion, I suggest adding clear function here, then items will be release when calling `g_array_free (split->tensorseg, TRUE);`.

```g_array_set_clear_func (split->tensorseg, g_free);```

Reply:
https://docs.gtk.org/glib/type_func.Array.set_clear_func.html

```g_array_set_clear_func (split->tensorseg, g_free);``` will do ```g_free (  tensor_dim ** )```, which is still the container of the element. I'll add a custom free function here.